letsencrypt/auth_handler.py

```diff
@@ -75,19 +75,34 @@ def get_authorizations(self, domains, best_effort=False):
 
         self._choose_challenges(domains)
 
+        failed_domains = set()
+
         # While there are still challenges remaining...
         while self.dv_c or self.cont_c:
             cont_resp, dv_resp = self._solve_challenges()
             logger.info("Waiting for verification...")
 
             # Send all Responses - this modifies dv_c and cont_c
-            self._respond(cont_resp, dv_resp, best_effort)
+            response = self._respond(cont_resp, dv_resp, best_effort)
+
+            if response:
+                failed_domains = failed_domains.union(response)
+
+        my_authzr = self.authzr
+
+        logger.debug("authzr: %s", my_authzr)
+
+        returnDomains = []
+        #Remove failing domains if best_effort is true
+        for domain in domains:
+            if not domain in failed_domains:
+                returnDomains.append(domain)
```

Comment:
this can be simplified as a list comprehension of the form
`returnDomains = [domain for domain in domains if domain not in failed_domains]`